cosmic-settings/src/pages/power/mod.rs

```diff
@@ -47,6 +53,45 @@ impl Page {
     }
 }
 
+fn battery_info() -> Section<crate::pages::Message> {
+    let descritpions = Slab::new();
+
+    Section::default()
+        .title(fl!("battery"))
+        .descriptions(descritpions)
+        .view::<Page>(move |_binder, _page, section| {
+            let runtime = tokio::runtime::Runtime::new().unwrap();
+            let battery = runtime.block_on(Battery::update_battery());
+
+            let battery_icon = widget::icon::from_name(battery.icon_name);
+            let battery_percent = widget::text(format!("{}%", battery.percent));
+
+            let remaining_time = |duration: Duration| {
+                let total_seconds = duration.num_seconds();
+
+                let hours = total_seconds / 3600;
+                let minutes = (total_seconds % 3600) / 60;
+                let seconds = total_seconds % 60;
+
+                fl!(
+                    "battery",
+                    "remaining-time",
+                    time = format!("{:02}:{:02}:{:02}", hours, minutes, seconds)
+                )
+            };
```

Comment:
It'd be more efficient to store this string in the page struct so it's not reallocated on every view update.